File: Lecture021 Solving Array Questions 2/RotateArray.py
def rotate(arr, k):

    temp=arr.copy()
    size = len(temp)
    for i in range(size):
        temp[(i+k) % size] = arr[i]

    return temp


if __name__ == '__main__':
    array = [11, 7, 3, 12, 4]
    ans = rotate(array,3)
    print("Rotated Array is ",ans)

File: Lecture021 Solving Array Questions 2/rotateArray.java
class rotateArray {
    public void rotate(int[] nums, int k) {
        int temp[] = new int[nums.length];
        for(int i =0;i<nums.length;i++){
            temp[((i+k)%nums.length)] = nums[i];
        }
        
        for(int j=0;j<nums.length;j++){
        nums[j] = temp[j];
      }
    }
}
